pkg/dev/dev.go

```diff
@@ -0,0 +1,128 @@
+// This file is part of MinIO Direct CSI
+// Copyright (c) 2020 MinIO, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package dev
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang/glog"
+)
+
+type BlockDevice struct {
+	Major   string
+	Minor   string
+	Devname string
+	Devtype string
+}
+
+func FindDrives(ctx context.Context) ([]BlockDevice, error) {
+	const head = "/sys/devices"
+	drives := []BlockDevice{}
+
+	if err := filepath.Walk(head, func(path string, info os.FileInfo, err error) error {
+		if ctx.Err() != nil {
+			return err
+		}
+		if strings.HasPrefix(info.Name(), "loop") {
+			return filepath.SkipDir
+		}
+		if info.Name() != "uevent" {
+			return nil
+		}
+		drive, err := parseUevent(path)
+		if err != nil {
+			glog.V(10).Info(err)
+			return nil
```

Comment:
Good that we dont error out on parse failure :+1

Reply:
Does this happen under normal operation or is it exceptional? If it's exceptional, then maybe log an `Error` instead of `Info`.

Reply:
It’s regular